command/agent/exec/exec.go

```diff
@@ -63,15 +64,11 @@ type Server struct {
 
 	childProcess      *child.Child
 	childProcessState childProcessState
+	childProcessLock  sync.Mutex
 
 	// exit channel of the child process
 	childProcessExitCh chan int
 
-	// we need to start a different go-routine to watch the
-	// child process each time we restart it.
-	// this function closes the old watcher go-routine so it doesn't leak
-	childProcessExitCodeCloser func()
-
```

Comment:
This closer was causing a race condition and complicating the code. It is not clear whether we actually need it -- the go routine it terminates is a simple forwarder, so it may be OK for it to be dangling.